lib/src/Uncertainty/Algorithm/MetaModel/Kriging/openturns/GaussianProcessFitter.hxx

```diff
@@ -0,0 +1,255 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief The class builds gaussian processes
+ *
+ *  Copyright 2005-2024 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_GAUSSIANPROCESSFITTER_HXX
+#define OPENTURNS_GAUSSIANPROCESSFITTER_HXX
+
+#include "openturns/MetaModelAlgorithm.hxx"
+#include "openturns/Basis.hxx"
+#include "openturns/CovarianceModel.hxx"
+#include "openturns/KrigingResult.hxx"
+#include "openturns/HMatrix.hxx"
+#include "openturns/OptimizationAlgorithm.hxx"
+#include "openturns/GaussianProcessFitterResult.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class GaussianProcessFitter
+ *
+ * The class fitting gaussian processes
+ */
+
+class OT_API GaussianProcessFitter
+  : public MetaModelAlgorithm
+{
+  CLASSNAME
+
+public:
+
+  enum LinearAlgebra { LAPACK, HMAT };
+
+  /** Default constructor */
+  GaussianProcessFitter();
+
+  /** Parameters constructor */
+  GaussianProcessFitter (const Sample & inputSample,
+                         const Sample & outputSample,
+                         const CovarianceModel & covarianceModel,
+                         const Bool keepCholeskyFactor = ResourceMap::GetAsBool("GaussianProcessFitter-KeepCovariance"));
+
+  GaussianProcessFitter (const Sample & inputSample,
+                         const Sample & outputSample,
+                         const CovarianceModel & covarianceModel,
+                         const Basis & basis,
```

Comment:
we could also use a default value "Basis(0)" for the basis argument so that only one ctor is needed